apps/internal/exported/exported.go

```diff
@@ -31,4 +31,6 @@ type TokenProviderResult struct {
 	AccessToken string
 	// ExpiresInSeconds is the lifetime of the token in seconds
 	ExpiresInSeconds int
+	// RefreshInSeconds indicates the suggested	time to refresh the token, if any
+	RefreshInSeconds int
```

Comment:
nit: this extensiblity point is really for Azure SDK only, so that they can plug in their MSI logic and benefit from MSAL caching. We don't have to add this here, but it's ok either way.

Reply:
But if you do add it, we need tests for it. And you need to plug it in.

Reply:
Sorry if I wasn't clear - pls remove it or add tests for this. But please reply to the comment to help me (as reviewer) know which path you chose.

Reply:
Apologies, I will be more prompt with the response on the comments here,
I am adding test for this.
We cannot remove this from exported as the same contract is used for `base.Client` as well as `Confidential.Client`